src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java

```diff
@@ -324,10 +325,18 @@ public final void markDirty() {
         return Optional.empty();
     }
 
+    // returns the amount of researches with at least 1 enabled item
+    private int nonEmptyResearches() {
+        return (int) Slimefun.getRegistry().getResearches()
+                .stream()
+                .filter(research -> research.getAffectedItems().stream().anyMatch(item -> item.getState() == ItemState.ENABLED))
+                .count();
+    }
```

Comment:
this can be static I believe, idk if it should be but it can be

Reply:
it should not be. It shouldn't be used outside this class.